README.md
# Device Creator

[![Device](https://cdn-images-1.medium.com/max/800/1*ddJUtOayzmuA0_Ynkgu9Jg.jpeg)](https://madeas.github.io/box-shadows/device/)

## About the project
**Device Creator** is a tool for creating simple and beautiful devices on your website. Created your device, edit the colors, background and add to it buttons. You can use the created devices to demonstrate your code, templates, pictures, adaptive design and much more.
By experimenting with buttons, you can use the tool to create, for example, the smartphone of your dreams. Toggle buttons, merge, change the size of the display, the color of the panel, etc. Do you dream of a pink apple? Please =)

[![Device](https://cdn-images-1.medium.com/max/800/1*8kSK3bYQ9SeK8PH9-i8EhQ.jpeg)](https://madeas.github.io/box-shadows/device/)

Entertainment can be endless. However, the main purpose of creating a tool is to use it on websites. Change a heavy image of a laptop or smartphone to several lines of code — this is a very big difference. And if the site-portfolio, where there are many such images? Many people are faced with the fact that the site has been downloaded for a very long time because of this … In addition, you can at any time make changes to the code.

Many people will say, “Wow, Fu, it’s easier to created himself” and in some ways it will be right. But I ask you not to judge strictly, the project is not positioned as a kind of super development. This a just fun and little support for the encoder) Now I’ll tell you everything.

You can use the created devices to demonstrate your code, templates, pictures, design created for mobile devices, placing html5-banners in it and much more. Ways of applying are just your imagination.

## Interface of the site
When you enter the site, you will see a small display around which actions will unfold. On the left and right side of the site there are buttons that change and add various parameters.

### Device
These are the main buttons, by the name of which it is not difficult to guess what type of device will be created. Currently, there are 5 types and device elements that are great for publishing on your site.

* phone-small
* phone-big
* tablet-portrait
* tablet-landscape
* ipod

### Background
Here are the buttons that change the color of the panel of your device. The default color of the panel is white, which can be changed to black or removed altogether. Below is the button for creating your own color.

* click on the circle near the button `← Choose`
* create your own color or enter the color code in the appeared window
* click to `Enter`

You can set the color values yourself. When the code is received, the values are reflected in the RGB format. If you need a hexadecimal representation, your HEX code will appear on the button next to the circle. Click on the button and copy the code.

### Box-shadow
The buttons of this block add shadows to your device, which makes it more realistic. There are 6 types + Shadow Removal button:

* bSd_light
* bSd_small
* bSd_classic
* bSd_toBlack
* bSd_around
* bSd_big
* bSd_none

### Display
This is not so important block, but if you want to create, for example, an empty panel for a smartphone, you can turn off the display by pressing the desired button. Likewise, if you are tired of starting text on the display, in the same block you can either stop it or completely turn it off.

### Create your own style
This is a special field in which you can enter any attribute and its value. To save multiple attributes, separate them with a semicolon.

[![Device](https://cdn-images-1.medium.com/max/800/1*lBBLP1k0Ao0OQ0EfD2dkWg.jpeg)](https://madeas.github.io/box-shadows/device/)

**Important!** When you edit this field, all previously created settings will be reset! Therefore, if you want to set, for example, your own sizes, you first need to enter them in this field, and then work with the buttons.

### Get code
When your work on creating the device is completed, click `Get code`. In the window that appears, you will see the code of your device. Copy the code by pressing the `Copy code` button and set to the desired place in your project.
You can easily add and change the styles of your device.

### Position
This block is not so important, but it can come in handy when you get the code. It presets the positioning parameter that you need to place on the site. I recommend using the buttons at the very last moment, ie, after all the work on setting up your device.

### Block flex
Select this to determine how the blocks within your device will be located.

### Buttons
When creating a device, you can experiment. Press the buttons one by one, turn them off or combine to create a suitable interface.
*Currently, the styles of the interface buttons are not displayed when you receive the code, but they are already in the plans for the near future.*



CHANGELOG.md
# Changes in the functionality of devices

### November 06, 2018

* The project is renamed to Device Creator.

### August 09, 2018

* Ability to change the appearance and position of the rear camera iSight. The view of the camera also varies depending on the angle of the device
* Added of visual buttons "menu and back", with the possibility of turning off / on

### August 07, 2018

* Add function 3d "Rotation device"
* Added types of background - Peach
* Improved visual mapping the camera, dynamics and the bottom buttons

### August 03, 2018

* Added 2 types of gradient background - Silver/Pink
* The developer buttons and the function for obtaining the device code are hidden - press the "I'm a developer"

### August 01, 2018

* Added the ability to paste stickers on the back panel of the deviсe - "Add sticker"
* Reset all settings button

### July 30, 2018

* Remote device type - iPod
* Added 3 types of shadows 

### July 28, 2018

* Ability to remove display, stop and remove text
* Rotate buttons in landscape
* Add bottom Circle button

### July 25, 2018

* Function to get device code - "Get code" button
* Now you can install your styles on the device - "Create your own style"
* Positiion - absolute/relative
* Display - flex/block
* Added back cover of the device
* Rotation function of the device by click

### July 24, 2018

* Public release
* 5 types of devices
* 4 colors devices and the ability to create your own color
* 5 types of shadows
* Buttons: Camera, Dynamic, 2 Bottom Line Buttons

[Top ↑][top]

[top]: #changes-in-the-functionality-of-devices


js/.js
// добавление класса
$(function() {
  var b = $('.btn_cls');
  b.click(function() {
    var cls = $(this).data("cls");
    $('#bSm').toggleClass(cls);
  });
});

// функция днд
$(function() {
  $('#dragshadow').draggable();
  $('#dragdisplay').draggable()
  $('#dragposition').draggable();
  $('#dragblock').draggable();
});

//добавление стилей из data-stl и выведение в поле #form_code
$('.btn').click(function() {
  $('#bSm').css($(this).data('stl'));

  var newStyles = $('#bSm').attr('style').replace(/; /g, ';\n');
  var val = $('#form_code').val();

  $('#form_code').val(val.replace(/#bSd \{([\s\S]*?)\}/, `#bSd {\n${newStyles}\n}`));
});

// type=color 
document.getElementById('bg_choose').onchange = e => {
  document.getElementById('choose').textContent = e.target.value;
  document.getElementById('bSm').style.backgroundColor = e.target.value;
  let ecls = document.querySelector('#bg_choose');
  ecls.dataset.cls = e.target.value;
}

// бегущая строка на дисплее
var typed = new Typed("#typed", {
  strings: [
    'Can place here:',
    'code <br /><pre><span class="pl-e">.display</span> {<br /><span class="pl-c1"><span class="pl-c1">  border-radius</span></span>: <span class="pl-c1">8<span class="pl-k">px</span></span>;<br /><span class="pl-c1"><span class="pl-c1">  min-height</span></span>: <span class="pl-c1">185<span class="pl-k">px</span></span>;<br /><span class="pl-c1"><span class="pl-c1">  padding</span></span>: <span class="pl-c1">15<span class="pl-k">px</span></span>;<br />}</pre>',
    'pictures,',
    'responsive design,',
    'and much more...',
  ],
  onStop: (arrayPos, self) => { // остановка текста по клику
    console.log('onStop', arrayPos, self);
  },
  onStart: (arrayPos, self) => { // старт текста по клику
    console.log('onStart', arrayPos, self)
  },
  onDestroy: (arrayPos, self) => { // удаление текста
    console.log('onDestroy', arrayPos, self)
  },
  onReset: (arrayPos, self) => { // сброс текста
    console.log('onReset', arrayPos, self)
  }
});

// стоп - старт по клику
var checkbox = document.querySelector("input[id=typstop]");
checkbox.addEventListener('change', function() {
  if (this.checked) {
    // Checkbox is checked..
    typed.stop();
  } else {
    // Checkbox is not checked..
    typed.start();
  }
});

// сброс - рестарт по клику
var checkbox = document.querySelector("input[id=typdest]");
checkbox.addEventListener('change', function() {
  if (this.checked) {
    // Checkbox is checked..
    typed.destroy();
  } else {
    // Checkbox is not checked..
    typed.reset();
  }
});

// генератор стилей
document.getElementById('adstl').onchange = e => {
  let ecls = document.querySelector('#adstl');
  ecls.dataset.cls = e.target.value;
  document.getElementById('bSm').style = e.target.value;
}

// добавление класса в дисплей
$(function() {
  var b = $('.btn_dis_cls');
  b.click(function() {
    var cls = $(this).data("cls");
    $('#display').toggleClass(cls);
  });
});

// функция копирования кода в буфер
var button = document.querySelector('.copy_code');
button.addEventListener('click', function(event) {
  var text = document.querySelector('#form_code');
  text.select();
  document.execCommand('copy');
});

// автосайз поля с кодом
var textarea = document.querySelector('.custom-textarea');
textarea.addEventListener('keydown', autosize);

function autosize() {
  var el = this;
  setTimeout(function() {
    el.style.cssText = 'height:' + el.scrollHeight + 'px';
  }, 0);
}

// добавление стикера на заднюю крышку
var back = document.querySelector("#back"),
  x = document.querySelector("#posX"),
  y = document.querySelector('#posY'),
  so = document.querySelector('#sizeOne'),
  st = document.querySelector('#sizeTwo'),
  si = document.querySelector('#sizeIn');
(style = () => {
  back.style.background = 'url(' + document.querySelector("#bgImage").value + ') ' + document.querySelector('[name="repeat"]:checked').parentNode.textContent + ' ' + x.value + '% ' + y.value + '%';
  // back.style.background = `url(${document.querySelector("#bgImage").value}) black ${document.querySelector('[name="repeat"]:checked').parentNode.textContent} ${x.value}% ${y.value}%`;
  back.style.backgroundSize = so.checked ? 'cover' : st.checked ? 'contain' : si.value + 'px';
})();
document.querySelectorAll('input').forEach(el => el.oninput = style);

// сброс стилей и кнопок
NodeList.prototype.forEach = Array.prototype.forEach;

document.getElementById('reset').addEventListener('click', function(e) {
  var sideLeft = document.getElementById('sidebar-left');
  var sideRight = document.getElementById('sidebar-right');
  var inputsL = sideLeft.querySelectorAll('input');
  var inputsR = sideRight.querySelectorAll('input');
  inputsL.forEach(function(el) {
    el.checked = false;
  });
  inputsR.forEach(function(el) {
    el.checked = false;
  });
  bSm.style = ''
});


js/other.js
// добавление класса
$(function() {
  var b = $('.btn_cls');
  b.click(function() {
    var cls = $(this).data("cls");
    $('#bSm').toggleClass(cls);
  });
});

// функция drag-and-drop
$(function() {
  $('#dragshadow').draggable();
  $('#dragdisplay').draggable()
  $('#dragposition').draggable();
  $('#dragblock').draggable();
  $('#dragbuttons').draggable();
  $('#dragcreate').draggable();
  $('#drag_back_set').draggable();
});

//добавление стилей из data-stl и выведение в поле #form_code
$('.btn').click(function() {
  $('#bSm').css($(this).data('stl'));
  var newStyles = $('#bSm').attr('style').replace(/; /g, ';\n');
  var val = $('#form_code').val();
  $('#form_code').val(val.replace(/#bSd \{([\s\S]*?)\}/, `#bSd {\n${newStyles}\n}`));
});

// type='color'
document.getElementById('bg_choose').onchange = e => {
  document.getElementById('choose').textContent = e.target.value;
  document.getElementById('bSm').style.background = e.target.value;
  let ecls = document.querySelector('#bg_choose');
  ecls.dataset.cls = e.target.value;
}

// бегущая строка typed display
var typed = new Typed("#typed", {
  strings: [
    'Can place here:',
    'code <br /><pre><span class="pl-e">.display</span> {<br /><span class="pl-c1"><span class="pl-c1">  border-radius</span></span>: <span class="pl-c1">8<span class="pl-k">px</span></span>;<br /><span class="pl-c1"><span class="pl-c1">  min-height</span></span>: <span class="pl-c1">185<span class="pl-k">px</span></span>;<br /><span class="pl-c1"><span class="pl-c1">  padding</span></span>: <span class="pl-c1">15<span class="pl-k">px</span></span>;<br />}</pre>',
    'pictures,',
    'responsive design,',
    'and much more...',
  ],
  onStop: (arrayPos, self) => { // остановка текста по клику
    console.log('onStop', arrayPos, self);
  },
  onStart: (arrayPos, self) => { // старт текста по клику
    console.log('onStart', arrayPos, self)
  },
  onDestroy: (arrayPos, self) => { // удаление текста
    console.log('onDestroy', arrayPos, self)
  },
  onReset: (arrayPos, self) => { // сброс текста
    console.log('onReset', arrayPos, self)
  }
});

// текст стоп/старт по клику
var checkbox = document.querySelector("input[id=typstop]");
checkbox.addEventListener('change', function() {
  if (this.checked) {
    // Checkbox is checked..
    typed.stop();
  } else {
    // Checkbox is not checked..
    typed.start();
  }
});

// текст сброс/рестарт по клику
var checkbox = document.querySelector("input[id=typdest]");
checkbox.addEventListener('change', function() {
  if (this.checked) {
    // Checkbox is checked..
    typed.destroy();
  } else {
    // Checkbox is not checked..
    typed.reset();
  }
});

// генератор стилей
document.getElementById('adstl').onchange = e => {
  let ecls = document.querySelector('#adstl');
  ecls.dataset.cls = e.target.value;
  document.getElementById('bSm').style = e.target.value;
}

// добавление классов в id='display'
$(function() {
  var b = $('.btn_dis_cls');
  b.click(function() {
    var cls = $(this).data("cls");
    $('#display').toggleClass(cls);
  });
});

// добавление классов в id='back'
$(function() {
  var b = $('.btn_back_cls');
  b.click(function() {
    var cls = $(this).data("cls");
    $('#back').toggleClass(cls);
  });
});

// сохранение кода в буфер обмена
var button = document.querySelector('.copy_code');
button.addEventListener('click', function(event) {
  var text = document.querySelector('#form_code');
  text.select();
  document.execCommand('copy');
});

// автосайз поля с кодом
var textarea = document.querySelector('.custom-textarea');
textarea.addEventListener('keydown', autosize);

function autosize() {
  var el = this;
  setTimeout(function() {
    el.style.cssText = 'height:' + el.scrollHeight + 'px';
  }, 0);
}

// добавление стикера на заднюю крышку
var back = document.querySelector('.back_cover'),
  x = document.querySelector('#posX'),
  y = document.querySelector('#posY'),
  so = document.querySelector('#sizeOne'),
  st = document.querySelector('#sizeTwo'),
  si = document.querySelector('#sizeIn');
(style = () => {
  back.style.backgroundImage = 'url(' + document.querySelector('#bgImage').value + ')';
  back.style.backgroundSize = so.checked ? 'cover' : st.checked ? 'contain' : si.value + 'px';
  back.style.backgroundRepeat = document.querySelector('[name="repeat"]:checked').parentNode.textContent;
  back.style.backgroundPosition = x.value + '%' + y.value + '%';
})();
document.querySelectorAll('input').forEach(el => el.oninput = style);

// сброс стилей и кнопок
NodeList.prototype.forEach = Array.prototype.forEach;

document.getElementById('reset').addEventListener('click', function(e) {
  var sideLeft = document.getElementById('sidebar-left');
  var sideRight = document.getElementById('sidebar-right');
  var inputsL = sideLeft.querySelectorAll('input[type=radio]');
  var inputsR = sideRight.querySelectorAll('input[type=radio]');
  inputsL.forEach(function(el) {
    el.checked = false;
  });
  inputsR.forEach(function(el) {
    el.checked = false;
  });
  bSm.style = '';
});

// добавить/убрать кнопки menu и back
var bi = document.getElementById('binone');
var bb = document.getElementById('back_button');
var mb = document.getElementById('menu_button');
bi.addEventListener('change', function() {
  if (this.checked) {
    bb.style.display = 'block';
    mb.style.display = 'block';
  } else {
    bb.style.display = 'none';
    mb.style.display = 'none';
  }
});

// открыть кнопки разработчика
var odb = document.querySelector('.opendev');
odb.onclick = e => {
  e.target.parentNode.style.display = 'none';
  var devb = document.querySelectorAll('.devbutton');
  document.querySelectorAll('.devbutton').forEach(n => n.classList.remove('devbutton'));
};
if (localStorage.getItem('an') == 1) opb.click();

// вращение устройства
var state = true;
document.querySelector('.anistop').onclick = () => {
  document.querySelector('.bsdisplay').style.animationPlayState = state ? 'running' : 'paused';
  state = !state;
};

// настройка камеры iSight
function toggle() {
  var dbblock = document.getElementById('drag_back_set');
  if (devplay.checked)
    dbblock.style.display = 'block';
  else
    dbblock.style.display = 'none'
}
var devplay = document.getElementById('device_play');
devplay.addEventListener('change', function() {
  toggle();
});
